net-misc/asterisk-opus/asterisk-opus-13.0.1.3.0.ebuild

```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit multilib
+
+DESCRIPTION="OPUS codec and supporting files for asterisk"
+HOMEPAGE="https://wiki.asterisk.org/wiki/display/AST/Codec+Opus"
+
+AST_PV="$(ver_cut 1-2)"
+MY_PV="$(ver_rs 2 _)"
+
+SRC_URI="amd64? (
+	https://downloads.digium.com/pub/telephony/codec_opus/asterisk-${AST_PV}/x86-64/codec_opus-${MY_PV}-x86_64.tar.gz
+)
+x86? (
+	https://downloads.digium.com/pub/telephony/codec_opus/asterisk-${AST_PV}/x86-32/codec_opus-${MY_PV}-x86_32.tar.gz
+)"
+
+LICENSE="Digium"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+
+RDEPEND="=net-misc/asterisk-${PV%%.*}*"
+
+RESTRICT="mirror strip"
+
+S="${WORKDIR}"
+
+pkg_setup() {
+	QA_FLAGS_IGNORED="/usr/$(get_libdir)/asterisk/modules/codec_opus.so"
+}
+
+src_install() {
+	local arch
+
+	if use x86; then
+		arch=x86_32
+	elif use amd64; then
+		arch=x86_64
+	else
+		die "Ebuild only functions for x86 and amd64."
+	fi
+
+	dodoc codec_opus-${MY_PV}-${arch}/README
+	insinto /usr/$(get_libdir)/asterisk/modules/
+	doins "codec_opus-${MY_PV}-${arch}/codec_opus.so"
+
+	insinto /var/lib/asterisk/documentation/thirdparty
+	doins "codec_opus-${MY_PV}-${arch}/codec_opus_config-en_US.xml"
```

Comment:
Why this weird directory?

Reply:
That's how they tar it up :).  Future proofing (hopefully) by using the versions and arch variables rather than hard coding.

Reply:
If you're referring to the thirdparty folder ... asterisk has build in documentation inside of it's CLI, in order for the opus code to load correctly, this .xml file has to be present in that folder.  Poorly enough, it has to be there during asterisk *startup*, not at module load time.  I consider that a bug but considering the amount of other stuff that's ongoing that I need to get sorted out for business that's the least of my worries.